### PR TITLE
chore: add migration for traffic data collection

### PR DESCRIPTION
**Pull request by daveleek**
## About the changes

Adds migration for creating table `stat_traffic_usage`.
This table primary-keys on day, traffic_group, and status_code_series.

Traffic group is the grouping for API endpoints for which traffic is counted.
status_code_series is 200/202 etc = 200, 304 etc = 300

### CODE REVIEW

File: src/migrations/20240208123212-create-stat-traffic-usage-table.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+exports.up = (db, callback) => {
+    db.runSql(
+        `
+        CREATE TABLE IF NOT EXISTS stat_traffic_usage(
+            day DATE NOT NULL,
+            traffic_group TEXT NOT NULL,
```
Comment (daveleek) [inline src/migrations/20240208123212-create-stat-traffic-usage-table.js:7]: ended up naming this traffic_group as the `group` name we initially came up with was a reserved keyword in Postgresql

File: src/migrations/20240208123212-create-stat-traffic-usage-table.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            status_code_series INT NOT NULL,
```
Comment (daveleek) [inline src/migrations/20240208123212-create-stat-traffic-usage-table.js:8]: appended a _series to make it clear that it's 200/300 etc, not 200/201/202/304 etc

**Comment by chriswk** [comment]
@gardleopard , @sjaanus do you think this would do for storing 400 days of data for visitation stats?

**Comment by chriswk** [comment]
I think creating subindexes on day, traffic_group and status_code_series could help. the PK index will be a thruple of (day, tg, scs), so if we want to do where status_code_series = 200 it will be a table scan.

**Comment by daveleek** [comment]
Thanks @chriswk - I've added the suggested indexes

**Review by chriswk — APPROVED** [review]
LGTM

**Pull request merged**